src/core/linalg/src/dense/4C_linalg_utils_densematrix_eigen.hpp

```diff
@@ -230,6 +230,102 @@ namespace Core::LinAlg
     return;
   }
 
+  /*!
+   * \brief Compute all (generally complex) generalized eigenvalues and right eigenvectors
+   *        of a pair of real, square, not necessarily symmetric matrices A and B.
+   *
+   * Solve the generalized eigenvalue problem
+   *
+   *    A * v = lambda * B * v
+   *
+   * where lambda = alpha / beta is the generalized eigenvalue returned by LAPACK GGEV.
+   *
+   * \note The eigenvalues are not sorted!
+   *
+   * \param A (in):  M-by-M matrix A
+   * \param B (in):  M-by-M matrix B
+   * \param S (out): M-by-M diagonal matrix holding the generalized eigenvalues
+   * \param V (out): M-by-M matrix whose columns are the generalized right eigenvectors
+   */
+  template <unsigned int dim>
+  void ggev(const Core::LinAlg::Matrix<dim, dim, double>& A,
+      const Core::LinAlg::Matrix<dim, dim, double>& B,
+      Core::LinAlg::Matrix<dim, dim, std::complex<double>>& S,
+      Core::LinAlg::Matrix<dim, dim, std::complex<double>>& V)
+  {
+    // ----- settings for generalized eigendecomposition ----- //
+
+    const char jobvl = 'N';  // do not compute left eigenvectors
+    const char jobvr = 'V';  // compute right eigenvectors
+
+    const int N = dim;
+
+    // Copy A and B since they will be overwritten by LAPACK
+    Matrix<dim, dim> tmpA(A.data(), false);
+    Matrix<dim, dim> tmpB(B.data(), false);
```

Comment:
```suggestion
    Matrix<dim, dim> tmpA = A;
    Matrix<dim, dim> tmpB = B;
```

Would be easier to read. Or is this potentially running into the view bug #492, @amgebauer?